draft-packages/popover/KaizenDraft/Popover/PopoverModern.tsx

```diff
@@ -0,0 +1,204 @@
+import { usePopper } from "react-popper"
+import { Icon } from "@kaizen/component-library"
+import closeIcon from "@kaizen/component-library/icons/close.icon.svg"
+
+import classNames from "classnames"
+import React, { useEffect, useState } from "react"
+import ReactDOM from "react-dom"
+import styles from "./styles.scss"
+import { Size, Variant } from "./types"
+import {
+  mapArrowVariantToClass,
+  mapLineVariant,
+  mapSizeToClass,
+  mapVariantToBoxClass,
+  mapVariantToIcon,
+  mapVariantToIconClass,
+} from "./classMappers"
+
+type Placement =
+  | "top"
+  | "bottom"
+  | "top-start"
+  | "top-end"
+  | "bottom-start"
+  | "bottom-end"
+
+export type ModernPopoverProps = {
+  readonly automationId?: string
+  readonly visible?: boolean
+  readonly onClose?: (event: React.MouseEvent<HTMLButtonElement>) => any
+  readonly variant?: Variant
+  readonly placement?: Placement
+  readonly size?: Size
+  readonly heading?: string
+  readonly dismissible?: boolean
+  readonly singleLine?: boolean
+  readonly children: React.ReactNode
+  /** For almost all intents and purposes, you should be using a pre-defined variant.
+   Please avoid using a custom icon unless you have a very good reason to do so. **/
+  readonly customIcon?: React.SVGAttributes<SVGSymbolElement>
+  readonly referenceElement: HTMLElement | null
+  /**
+   * Render the tooltip inside a react portal, given the ccs selector.
+   * This is typically used for instances where the menu is a descendant of an
+   * `overflow: scroll` or `overflow: hidden` element.
+   */
+  readonly portalSelector?: string
```

Comment:
To summarise the changes to these props compared to the legacy ones:
* new prop `portalSelector`
* new prop `referenceElement`
* `position` and `side` has been merged into the new prop `placement`